cmd/kube-scheduler/app/options/options.go

```diff
@@ -163,9 +163,6 @@ func (o *Options) ApplyLeaderElectionTo(cfg *kubeschedulerconfig.KubeSchedulerCo
 	if leaderelection.Changed("leader-elect-retry-period") {
 		cfg.LeaderElection.RetryPeriod = o.LeaderElection.RetryPeriod
 	}
-	if leaderelection.Changed("leader-elect-resource-lock") {
-		cfg.LeaderElection.ResourceLock = o.LeaderElection.ResourceLock
-	}
```

Comment:
revert this change, we still want to use the config field

instead, validate in ValidateKubeSchedulerConfiguration that the field is set to `leases` and fail if it isn't (so a config file using some other lease resource type fails instead of silently ignoring the now-unsupported lease type)